x-pack/plugins/index_management/public/sections/index_list/components/index_actions_context_menu/index_actions_context_menu.js

```diff
@@ -62,26 +62,24 @@ class IndexActionsContextMenuUi extends Component {
     const allOpen = all(indexNames, indexName => {
       return indexStatusByName[indexName] === INDEX_OPEN;
     });
-    const oneIndexSelected = this.oneIndexSelected();
-    const entity = this.getEntity(oneIndexSelected);
-    const entityUpper = `${entity[0].toUpperCase()}${entity.slice(1)}`;
+    const selectedIndexCount = indexNames.length;
     const items = [];
-    if (!detailPanel && oneIndexSelected) {
+    if (!detailPanel && selectedIndexCount === 1) {
       items.push({
         name: intl.formatMessage({
-          id: 'xpack.idxMgmt.indexActionsMenu.showEntitySettingsLabel',
-          defaultMessage: 'Show {entity} settings',
-        }, { entity }),
+          id: 'xpack.idxMgmt.indexActionsMenu.showIndexSettingsLabel',
+          defaultMessage: '{selectedIndexCount, plural, one {Show index settings} other {Show indices settings} }',
```

Comment:
I think it might make this a bit more maintainable if we only pluralize the noun; this way someone who is scanning the string doesn't need to compare the singular and plural forms for semantic equality, just for correct pluralization. What do you think?

```js
name: intl.formatMessage({
   id: 'xpack.idxMgmt.indexActionsMenu.showIndexSettingsLabel',
   defaultMessage: 'Show {selectedIndexCount, plural, one {index} other {indices} } settings',
}, { selectedIndexCount }),
```

Reply:
You are right! I will fix it.